velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -554,4 +554,34 @@ struct ArrayTrimFunctionString {
   }
 };
 
+/// This class implements the array flatten function.
+///
+/// DEFINITION:
+/// flatten(x) → array
+/// Flattens an array(array(T)) to an array(T) by concatenating the contained
+/// arrays.
+template <typename T>
+struct ArrayConcatFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T)
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Array<Generic<T1>>>& out,
+      const arg_type<Array<Array<Generic<T1>>>>& arrays) {
+    int64_t elementCount = 0;
+    for (const auto& array : arrays) {
+      elementCount += array.value().size();
+    }
+    out.reserve(elementCount);
+    for (const auto& array : arrays) {
+      for (auto element : array.value()) {
+        if (element.has_value()) {
+          auto& newItem = out.add_item();
+          newItem.copy_from(element.value());
+        } else {
+          out.add_null();
```

Comment:
This is only one layer? Not recursive flatten?

Reply:
Yes only one layer:

Run the following query in Presto:
```
SELECT flatten(ARRAY[ARRAY[ARRAY[1, 1], ARRAY[2, 2]], ARRAY[ARRAY[1, 1], ARRAY[2, 2]]])
```

Result:
```
[[1, 1], [2, 2], [1, 1], [2, 2]]
```